src/test/java/ch/uzh/ifi/hase/soprafs21/controller/UsersApiControllerTest.java

```diff
@@ -130,4 +131,76 @@ void logoutNullObject () throws Exception{
     }
 
 
+
+    @Test
+    void getUserSuccess() throws Exception {
+        UserDto mockedUserDto = getMockedUserDto();
+        given(userService.getUserDetails(Mockito.any())).willReturn(mockedUserDto);
+        ResponseEntity<UserDto> responseEntity = usersApiController.usersUserIdGet(1L);
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+        assertEquals(mockedUserDto,responseEntity.getBody());
+    }
+
+    @Test
+    void getUserNotFound() throws Exception {
+        UserDto mockedUserDto = getMockedUserDto();
+
+        given(userService.getUserDetails(Mockito.any())).willReturn(null);
+
+        Exception exception = assertThrows(ResponseStatusException.class, () -> {
+            usersApiController.usersUserIdGet(1L);
+        });
+
+        String expectedMessage = "User not found";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    void getUserNullId() throws Exception {
+        UserDto mockedUserDto = getMockedUserDto();
+
+        given(userService.getUserDetails(Mockito.any())).willReturn(null);
+
+        Exception exception = assertThrows(ResponseStatusException.class, () -> {
+            usersApiController.usersUserIdGet(null);
+        });
+
+        String expectedMessage = "Invalid input userId cannot be null";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    private UserDto getMockedUserDto() {
```

Comment:
You can use Mockito to instantiate a mocked userDTO

Reply:
removed this piece of code as service is updated to return entity instead of dto object.